src/datahike/db.cljc

```diff
@@ -677,6 +656,58 @@
 
 ;; ----------------------------------------------------------------------------
 
+(def db-caches (atom {}))
+
+(defn insert-cache [db-hash cache]
+  (let [caches @db-caches]
+    (if (>= (count caches) max-db-caches)
+      (let [[last-used-db _] (reduce (fn [[d-hash1 [c1 ^Date last-accessed1]]
```

Comment:
It would be better to use a LRU cache here, we use `clojure.core.cache` everywhere else, have you taken a look at that? Your reduce takes linear time in the size of the cache.

Reply:
Hadn't heard of that `clojure.core.cache` yet. Will have a look if at if is usable in this case